backend/src/services/userService.ts

```diff
@@ -90,6 +100,30 @@ const checkPassword = async (
     }
 };
 
+const changePassword = async (
+    userID: string,
+    currentPassword: string,
+    newPassword: string
+): Promise<RequestInfo> => {
+    const hash = await selectPasswordbyID(userID);
+    if (hash === null) return DEFAULT_ERROR;
+    try {
+        if (await bcrypt.compare(currentPassword, hash)) {
```

Comment:
We don't want to call bcrypt.compare here since we already have checkPassword function that takes the user name and password.